crates/pallets/starknet/src/lib.rs

```diff
@@ -376,10 +377,26 @@ pub mod pallet {
             }
 
             for (sierra_class_hash, casm_class_hash) in self.sierra_to_casm_class_hash.iter() {
+                // ContractClasses can be indexed both by Sierra and Casm hashes, so both should be checked
+                let is_sierra_hash_in_contracts = ContractClasses::<T>::contains_key(sierra_class_hash);
+                let is_casm_hash_in_contracts = ContractClasses::<T>::contains_key(casm_class_hash);
```

Comment:
Only `is_sierra_hash_in_contracts` should be true. 
`is_casm_hash_in_contracts` is true we have a problem: we are storing a cairo1 contract in `ContractClasses` by its casm class instead of its sierra one.
This should not happen.

Reply:
Okay I understand it now. Fixed it.

Reply:
@tdelabro Can you check it now, please?